### PR TITLE
Add Protobuf models

### PR DESCRIPTION
**Pull request by adelowo**
This PR introduces Protobuf into the model and removes the usage of `interface` on the models.

> I removed the invalid chain ID test as big.Int by default will always convert -ve numbers to positives when using `Bytes()`, `String()` and `Int64()` so that test case not so useful for us

### CODE REVIEW

File: proto/v1/asset.proto [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  oneof to {
+    AssetType to_asset = 5;  // The asset being received.
+    StakeType to_stake = 6;  // The stake being received.
+    LoanType to_loan = 7;    // The loan being received.
+  }
+
+  ExtraData extra_data = 8;  // Additional data for the intent.
+  ProcessingStatus status = 9;  // The processing status of the intent.
+  int64 created_at = 10 [(buf.validate.field).int64 = {gt: 0}];  // The creation timestamp of the intent.
+  int64 expiration_at = 11 [(buf.validate.field).int64 = {gte: 0}];  // The expiration timestamp of the intent.
```
Comment (hpsing) [inline proto/v1/asset.proto]: use `google.protobuf.timestamp`

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bufbuild/buf-setup-action@v1
+
+      - name: Update dependencies
+        run: buf mod update
+
+      - run: buf lint
+        name: Run Linter
```
Comment (hpsing) [inline .github/workflows/lint.yml]: name then run command for better readability

File: proto/v1/asset.proto (hunk 1 above)
Comment (adelowo) [reply proto/v1/asset.proto]: On that already. Broke a few tests. Fixing

File: .github/workflows/bob.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+name: Bob build
+on:
```
Comment (hpsing) [inline .github/workflows/bob.yml:2]: create a `github action` to publish ts-gen in `npm` for the `frontend` and `JS-SDK` (use a consistance package version)

File: README.md [hunk 4]
```diff
@@ -2,6 +2,16 @@
 
 Reusing data types, domain model or `Shared Kernel` in DDD parlance.
```
Comment (hpsing) [inline README.md:6]: add docs how new use case and further extendebility should occur

File: intents_test.go [hunk 5]
```diff
@@ -2,40 +2,54 @@ package model
...
 		return
 	}
-	// Validate the kind-specific fields
-	for _, intent := range body.Intents {
-		if err := intent.ValidateIntent(); err != nil {
-			c.JSON(http.StatusBadRequest, gin.H{"error": "The Intent's kind does not validate"})
-			return
-		}
+
+	if err := v.Validate(&body); err != nil {
```
Comment (hpsing) [inline intents_test.go:44]: 👍🏼

**Review by hpsing — APPROVED** [review]
LGTM! 💯

**Review by hpsing — APPROVED** [review]

File: bigint.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// ToBigInt converts a protobuf BigInt message to a *big.Int.
+func ToBigInt(b *protov1.BigInt) (*big.Int, error) {
+	result := new(big.Int)
+	result = result.SetBytes(b.GetValue())
+
+	if len(result.Bits()) == 0 {
+		return nil, errors.New("amount cannot be a zero amount")
+	}
```
Comment (blewater) [inline bigint.go:17]: @adelowo, Why can it not be zero? Like gas of 0 or default values?

Comment (adelowo) [reply bigint.go:17]: Because we only use BigInt for the TX amount and chain ID. A zero chain id is invalid. Also, a zero amount for a TX is invalid too.

**Pull request merged**